File: backend/app/schemas/conversation.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.conversation_log import ConversationRole


class ChatRequest(BaseModel):
    message: str


class ChatResponse(BaseModel):
    reply: str
    intent_json: dict | None = None
    actions_taken: list[str] = []


class ConversationLogResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    role: ConversationRole
    content: str
    intent_json: dict | None
    created_at: datetime

File: frontend/src/types/conversation.ts
export type ConversationRole = 'hunter' | 'assistant'

export interface ChatRequest {
  message: string
  thread_id?: string | null
}

export interface ChatResponse {
  reply: string
  intent_json: Record<string, unknown> | null
  actions_taken: string[]
}

export interface ConversationLogResponse {
  id: string
  project_id: string
  thread_id: string | null
  role: ConversationRole
  content: string
  intent_json: Record<string, unknown> | null
  created_at: string
}
